### DISCUSSION TITLE
Can we replace the renderer with a React one?

### DISCUSSION THREAD

**Discussion by sudo-sand** [freshframework/fresh#598] [Q&A] [answered]
Is it possible to use react instead of preact?

**Comment by predaytor** [comment]
I do not think so. Preact, as I understand it, was the choice for the whole framework (islands) architecture.

**Reply by shishkin** [reply]
I don't think Preact is key to islands. Astro for example does islands with a broad choice of frameworks, including React.

But what would be a reason to insist on React over Preact (with compat)?

**Comment by marvinhagemeister** [comment, marked as answer]
It's unlikely that we'll change the underlying framework used in Fresh anytime soon. It currently relies on a bunch of Preact APIs that are not available in other frameworks. The way Astro does it is by having their own compiler on top of everything which is certainly pretty cool, but comes with it's own set of tradeoffs.
